src/protocols/secure_channel/CASESession.cpp

```diff
@@ -695,16 +681,15 @@ CHIP_ERROR CASESession::HandleSigmaR2(System::PacketBufferHandle & msg)
 
     SuccessOrExit(err = decryptedDataTlvReader.Next());
     VerifyOrExit(TLV::TagNumFromTag(decryptedDataTlvReader.GetTag()) == ++decodeTagIdSeq, err = CHIP_ERROR_INVALID_TLV_TAG);
-    responderOpCertLen = static_cast<uint16_t>(decryptedDataTlvReader.GetLength());
-    SuccessOrExit(err = decryptedDataTlvReader.GetBytes(responderOpCert, responderOpCertLen));
+    responderOpCertLen = static_cast<size_t>(decryptedDataTlvReader.GetLength());
+    SuccessOrExit(err = decryptedDataTlvReader.GetBytes(responderOpCert, sizeof(responderOpCert)));
```

Comment:
This is complicated enough that it might be worthwhile to add a comment here pointing out that this is where we are checking that `responderOpCertLen <= sizeof(responderOpCert)`.

Reply:
Done